public/app/features/provisioning/Config/ConfigForm.tsx

```diff
@@ -65,6 +67,25 @@ export function ConfigForm({ data }: ConfigFormProps) {
   const localFields = type === 'local' ? getLocalProviderFields(type) : null;
   const hasTokenInstructions = getHasTokenInstructions(type);
 
+  // TODO: this should be removed after 12.2 is released
+  const checkTokenMigrationWarning = useCallback(() => {
+    if (data?.spec?.type !== 'local' && !data?.secure?.token) {
+      setTokenConfigured(false);
+      setError(
+        'token',
+        {
+          type: 'manual',
+          message: `Enter your ${gitFields?.tokenConfig.label ?? 'access token'}`,
+        },
+        { shouldFocus: true }
+      ); // does not seem to focus
+    }
+  }, [data, setTokenConfigured, setError, gitFields]);
+
+  useEffect(() => {
+    checkTokenMigrationWarning();
+  }, [checkTokenMigrationWarning]);
```

Comment:
```suggestion
useEffect(() => {
  // TODO: this should be removed after 12.2 is released
  if (data?.spec?.type !== 'local' && !data?.secure?.token) {
    setTokenConfigured(false);
    setError(
      'token',
      {
        type: 'manual',
        message: `Enter your ${gitFields?.tokenConfig.label ?? 'access token'}`,
      },
      { shouldFocus: true }
    ); // does not seem to focus
  }
}, [data, gitFields, setTokenConfigured, setError]);
```

I think this can be done inline without declaring a separate function.